File: Ejercicios/01-FindNeedle/FindNeedle.rb
def FindNeedle (haystack, needle)

    haystack.split("").each.with_index do |hayValue, hayIndex|
        needle.split("").each.with_index do |needleValue, needleIndex|
            if haystack[hayIndex + needleIndex] != needleValue
                break
            end
            if needleIndex == needle.length - 1 
                return hayIndex
            end
        end
    end
    return -1
end


puts FindNeedle("react-redux", "redux") 
# 6

puts FindNeedle("pinky", "puntual")
# -1

puts FindNeedle("rereredux", "reredux")
# 2

File: Ejercicios/01-FindNeedle/FindNeedle.js
function FindNeedle(haystack, needle) {
    for (let i = 0; i < haystack.length - needle.length + 1; i++) {
        if (haystack.slice(i, i + needle.length) === needle) {
            console.log(
                `${haystack} -> ${needle} \n` +
                    haystack.slice(0, i) +
                    "(" +
                    needle.toUpperCase() +
                    ")" +
                    haystack.slice(i + needle.length)
            )
            return i
        }
    }
    console.log(`${haystack} -> ${needle} \n-1`)
    return -1
}

module.exports = FindNeedle
